docs/maui/essentials/speech-to-text.md

```diff
@@ -85,12 +85,48 @@ async Task Listen(CancellationToken cancellationToken)
 }
 ```
 
+or using events:
+
+```csharp
+async Task Listen(CancellationToken cancellationToken)
+{
+    var isGranted = await speechToText.RequestPermissions(cancellationToken);
+    if (!isGranted)
+    {
+        await Toast.Make("Permission not granted").Show(CancellationToken.None);
+        return;
+    }
+
+    speechToText.RecognitionResultUpdated += OnRecognitionTextUpdated;
+    speechToText.RecognitionResultCompleted += OnRecognitionTextCompleted;
+    await SpeechToText.StartListenAsync(CultureInfo.CurrentCulture, CancellationToken.None);
+    
+    // Some logic
+    
+    await SpeechToText.StopListenAsync(CancellationToken.None);
+    SpeechToText.Default.RecognitionResultUpdated -= OnRecognitionTextUpdated;
+    SpeechToText.Default.RecognitionResultCompleted -= OnRecognitionTextCompleted;
+
+    void OnRecognitionTextUpdated(object? sender, SpeechToTextRecognitionResultUpdatedEventArgs args)
+    {
+        RecognitionText += args.RecognitionResult;
+    };
+    
+    void OnRecognitionTextCompleted(object? sender, SpeechToTextRecognitionResultCompletedEventArgs args)
+    {
+        RecognitionText = args.RecognitionResult;
+    };
+}
+```
```

Comment:
What do you think about splitting this code into two methods, `async Task StartListen()` and `async Task `StopListen()`. 

My assumption is that devs who use it this way will have two separate buttons (or two separate `ICommand`s) to start/stop listening.

We created an example of using two buttons in the sample app too! https://github.com/CommunityToolkit/Maui/blob/09ca1e9fc598da198e8a45f2e859b939e7b943cd/samples/CommunityToolkit.Maui.Sample/ViewModels/Essentials/SpeechToTextViewModel.cs#L123-L161